packages/tupaia-web/src/layout/UserMenu/UserInfo.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { Link, LinkProps } from 'react-router-dom';
+import { OutlinedButton, TextButton } from '@tupaia/ui-components';
+import { USER_ROUTES } from '../../Routes';
+
+/**
+ * UserInfo is a component that displays the user's name if user is logged in, or a register and sign in button if not set
+ */
+const Wrapper = styled.div`
+  @media screen and (max-width: ${({ theme }) => theme.breakpoints.values.md}px) {
+    display: none;
+  }
+`;
+
+const UsernameContainer = styled.p<{
+  $isLandingPage?: boolean;
+}>`
+  padding-right: 1.5rem;
+  padding-right: ${({ $isLandingPage }) => ($isLandingPage ? '1.5rem' : '5px')};
+  margin: 0;
+  font-weight: ${({ $isLandingPage, theme }) =>
+    $isLandingPage ? theme.typography.fontWeightMedium : theme.typography.fontWeightRegular};
+  font-size: 0.875rem;
+  text-transform: ${({ $isLandingPage }) => ($isLandingPage ? 'uppercase' : 'none')};
+  @media screen and (max-width: ${({ theme }) => theme.breakpoints.values.md}px) {
+    display: none;
+  }
+`;
+
+const Register = styled(TextButton).attrs({
+  component: Link,
+})`
```

Comment:
```suggestion
const Register = styled(Link).attrs({
  component: TextButton,
})`
```

There is a ts lint error where it complains about the `to` prop but switching the 2 components around fixed it for me. Also if we have the login and registers as modals we need to pass in the state: { location } so that the router knows how to display them. I created a component called RouterButton to save repeating that which is in the login branch so if I merge that in first we could re-use that here